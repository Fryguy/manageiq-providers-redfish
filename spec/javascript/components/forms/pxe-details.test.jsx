import PxeDetails from '../../../../app/javascript/components/forms/pxe-details'

let renderComponent;
let physicalServers = [1, 2, 3];
let pxeServers = [{value: 'pxe1', label: 'PXE1'}];
let pxeImages = [{value: 'img1', label: 'IMG1'}];
let templates = [{value: 'templ1', label: 'TEMPL1'}];

describe('PxeDetails', () => {
  beforeAll(() => {
    renderComponent = (loading = false, physicalServers = [], pxeServers = [], pxeImages = [], templates = []) => shallowRedux(
      <PxeDetails
        updateFormState={jest.fn()} loading={loading}
        physicalServerIds={physicalServers}
        pxeServers={pxeServers}
        pxeImages={pxeImages}
        customizationTemplates={templates}
      />
      );
  });

  describe('renders', () => {
    it('spinner', () => {
      let component = renderComponent(true);
      expect(toJson(component)).toMatchSnapshot();
    });

    describe('form', () => {
      it('without pxe servers', () => {
        let component = renderComponent(false, physicalServers);
        expect(toJson(component)).toMatchSnapshot();
      });

      it('with pxe servers', () => {
        let component = renderComponent(false, physicalServers, pxeServers);
        expect(toJson(component)).toMatchSnapshot();
      });

      it('with pxe servers and pxe images', () => {
        let component = renderComponent(false, physicalServers, pxeServers, pxeImages);
        expect(toJson(component)).toMatchSnapshot();
      });

      it('with pxe servers and pxe images and templates', () => {
        let component = renderComponent(false, physicalServers, pxeServers, pxeImages, templates);
        expect(toJson(component)).toMatchSnapshot();
      });
    });
  });
});
