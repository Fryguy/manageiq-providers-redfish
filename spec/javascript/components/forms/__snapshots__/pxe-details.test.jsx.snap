// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PxeDetails renders form with pxe servers 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={true}
  inline={false}
>
  <FormSpy
    onChange={[Function]}
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <h2>
          _Number of servers to be provisioned: 3_
        </h2>
      </Col>
    </Row>
    <hr />
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          inputColumnSize={8}
          label="_PXE Server_"
          labelColumnSize={3}
          name="pxeServer"
          options={
            Array [
              Object {
                "label": "PXE1",
                "value": "pxe1",
              },
            ]
          }
          parse={[Function]}
          placeholder="_Select a PXE Server_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          disabled={true}
          format={[Function]}
          inputColumnSize={8}
          label="_PXE Image_"
          labelColumnSize={3}
          name="pxeImage"
          options={Array []}
          parse={[Function]}
          placeholder="_Select a PXE Image_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          disabled={true}
          format={[Function]}
          inputColumnSize={8}
          label="_Customization Template_"
          labelColumnSize={3}
          name="customizationTemplate"
          options={Array []}
          parse={[Function]}
          placeholder="_Select Customization Template_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <hr />
    </Row>
  </Grid>
</Form>
`;

exports[`PxeDetails renders form with pxe servers and pxe images 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={true}
  inline={false}
>
  <FormSpy
    onChange={[Function]}
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <h2>
          _Number of servers to be provisioned: 3_
        </h2>
      </Col>
    </Row>
    <hr />
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          inputColumnSize={8}
          label="_PXE Server_"
          labelColumnSize={3}
          name="pxeServer"
          options={
            Array [
              Object {
                "label": "PXE1",
                "value": "pxe1",
              },
            ]
          }
          parse={[Function]}
          placeholder="_Select a PXE Server_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          disabled={false}
          format={[Function]}
          inputColumnSize={8}
          label="_PXE Image_"
          labelColumnSize={3}
          name="pxeImage"
          options={
            Array [
              Object {
                "label": "IMG1",
                "value": "img1",
              },
            ]
          }
          parse={[Function]}
          placeholder="_Select a PXE Image_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          disabled={true}
          format={[Function]}
          inputColumnSize={8}
          label="_Customization Template_"
          labelColumnSize={3}
          name="customizationTemplate"
          options={Array []}
          parse={[Function]}
          placeholder="_Select Customization Template_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <hr />
    </Row>
  </Grid>
</Form>
`;

exports[`PxeDetails renders form with pxe servers and pxe images and templates 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={true}
  inline={false}
>
  <FormSpy
    onChange={[Function]}
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <h2>
          _Number of servers to be provisioned: 3_
        </h2>
      </Col>
    </Row>
    <hr />
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          inputColumnSize={8}
          label="_PXE Server_"
          labelColumnSize={3}
          name="pxeServer"
          options={
            Array [
              Object {
                "label": "PXE1",
                "value": "pxe1",
              },
            ]
          }
          parse={[Function]}
          placeholder="_Select a PXE Server_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          disabled={false}
          format={[Function]}
          inputColumnSize={8}
          label="_PXE Image_"
          labelColumnSize={3}
          name="pxeImage"
          options={
            Array [
              Object {
                "label": "IMG1",
                "value": "img1",
              },
            ]
          }
          parse={[Function]}
          placeholder="_Select a PXE Image_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          disabled={false}
          format={[Function]}
          inputColumnSize={8}
          label="_Customization Template_"
          labelColumnSize={3}
          name="customizationTemplate"
          options={
            Array [
              Object {
                "label": "TEMPL1",
                "value": "templ1",
              },
            ]
          }
          parse={[Function]}
          placeholder="_Select Customization Template_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <hr />
    </Row>
  </Grid>
</Form>
`;

exports[`PxeDetails renders form without pxe servers 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={true}
  inline={false}
>
  <FormSpy
    onChange={[Function]}
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <h2>
          _Number of servers to be provisioned: 3_
        </h2>
      </Col>
    </Row>
    <hr />
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          inputColumnSize={8}
          label="_PXE Server_"
          labelColumnSize={3}
          name="pxeServer"
          options={Array []}
          parse={[Function]}
          placeholder="_Select a PXE Server_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          disabled={true}
          format={[Function]}
          inputColumnSize={8}
          label="_PXE Image_"
          labelColumnSize={3}
          name="pxeImage"
          options={Array []}
          parse={[Function]}
          placeholder="_Select a PXE Image_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          disabled={true}
          format={[Function]}
          inputColumnSize={8}
          label="_Customization Template_"
          labelColumnSize={3}
          name="customizationTemplate"
          options={Array []}
          parse={[Function]}
          placeholder="_Select Customization Template_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <hr />
    </Row>
  </Grid>
</Form>
`;

exports[`PxeDetails renders spinner 1`] = `
<div
  className="spinner spinner-lg"
/>
`;
